### PR TITLE
feat: enhance createContainer API with missing parameters

### LINKED ISSUE #6010: Extent createContainer Podman Desktop API with missing parameters
### Is your enhancement related to a problem? Please describe

After adding HealthCheck support (4bc1035), I noticed a lots of parameters are missing.

### Describe the solution you'd like

Add the missing parameters

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jeffmaury**
Fixes #6010

### What does this PR do?

Extend createContainer API

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

#6010 

### How to test this PR?

Unit tests or launch a container

### CODE REVIEW

File: packages/extension-api/src/extension-api.d.ts [hunk 1]
```diff
@@ -2256,6 +2256,11 @@ declare module '@podman-desktop/api' {
     Retries?: number;
   }
 
+  interface NetworkingConfig {
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    EndpointsConfig?: { [key: string]: {} };
```
Comment (benoitf) [inline packages/extension-api/src/extension-api.d.ts]: should we rather fix the definition ?

**Review by cdrage — APPROVED** [review]

**Review by cdrage — APPROVED** [review]
After opening up #6035 I did not realize we already have a PR!

I was able to test this and successfully passed in platform no problem in order to get my container to build on arm64.

LGTM code wise.

**Review by benoitf — APPROVED** [review]

**Review by benoitf — DISMISSED** [review]
I still see som ban-types usage

File: packages/extension-api/src/extension-api.d.ts [hunk 2]
```diff
@@ -2378,6 +2482,58 @@ declare module '@podman-desktop/api' {
...
+     *
+     * Command is already escaped (Windows only)
+     */
+    ArgsEscaped?: boolean;
+
+    /**
+     * An object mapping mount point paths inside the container to empty objects.
+     */
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    Volumes?: { [volume: string]: {} };
```
Comment (benoitf) [inline packages/extension-api/src/extension-api.d.ts]: can we remove all the ban-types ?

File: packages/main/src/plugin/api/container-info.ts [hunk 3]
```diff
@@ -130,6 +213,17 @@ export interface ContainerCreateOptions {
   Detach?: boolean;
   start?: boolean;
   HealthCheck?: HealthConfig;
+  ArgsEscaped?: boolean;
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  Volumes?: { [volume: string]: {} };
```
Comment (benoitf) [inline packages/main/src/plugin/api/container-info.ts]: can we remove all the ban-types ?

File: packages/extension-api/src/extension-api.d.ts (hunk 2 above)
Comment (jeffmaury) [reply packages/extension-api/src/extension-api.d.ts]: I don't know which type as the Docker reference does not detail it

https://docs.docker.com/engine/api/v1.44/#tag/Container/operation/ContainerCreate

Comment (benoitf) [reply packages/extension-api/src/extension-api.d.ts]: if we don't know should it be `unknown` ?

**Review by deboer-tim — APPROVED** [review]

**Pull request merged**